server/src/ClientToDBAPI/UsersRoute/mainRoute.ts

```diff
@@ -1,32 +1,62 @@
-    .then((result:any) => {
-        if (result.data?.userById)
-            response.send(result.data?.userById);
-        else
-            response.status(400).send(`Couldn't find the user nor get its status`);
-    })
-    .catch(error => response.status(500).send('Error while trying to fetch isActive user status'))
+    graphqlRequest(GET_IS_USER_ACTIVE, response.locals, { id: response.locals.user.id })
+        .then((result: any) => {
+            if (result.data?.userById)
+                response.send(result.data?.userById);
+            else
+                response.status(400).send(`Couldn't find the user nor get its status`);
+        })
+        .catch(error => response.status(500).send('Error while trying to fetch isActive user status'))
 })
 
 usersRoute.post('/updateIsUserActive', (request: Request, response: Response) => {
-    graphqlRequest(UPDATE_IS_USER_ACTIVE, response.locals, {id: response.locals.user.id, isActive: request.body.isActive})
-       .then((result:any) => {
-        if(result.data.updateUserById)
-            response.send(result.data.updateUserById.user);
-        else
-            response.status(400).send(`Couldn't find the user nor update the status`);
-
-       })
-       .catch(error => response.status(500).send('Error while trying to activate / deactivate user'))
+    graphqlRequest(UPDATE_IS_USER_ACTIVE, response.locals, { id: response.locals.user.id, isActive: request.body.isActive })
+        .then((result: any) => {
+            if (result.data.updateUserById)
+                response.send(result.data.updateUserById.user);
+            else
+                response.status(400).send(`Couldn't find the user nor update the status`);
+
+        })
+        .catch(error => response.status(500).send('Error while trying to activate / deactivate user'))
 })
 
+usersRoute.get('/getUserById', (request: Request, response: Response) => {
+    graphqlRequest(GET_USER_BY_ID, response.locals, { id: request.query.id })
+        .then((result: any) => response.send(result.data));
+});
+
+usersRoute.post('/changeInvestigator', (request: Request, response: Response) => {
+    graphqlRequest(UPDATE_INVESTIGATOR, response.locals, { id: request.query.id })
+        .then((result: any) => response.send(result.data));
+});
+
+usersRoute.get('/getAllGroupUsers', (request: Request, response: Response) => {
+    graphqlRequest(GET_ALL_GROUP_USERS, response.locals, { investigationGroup: +request.query.investigationGroup })
+        .then((result: any) => {
+            let users: User[] = [];
+            if (result && result.data && result.data.allUsers) {
+                users = result.data.allUsers.nodes.map((user: User) => ({
+                    ...user,
+                    token: ''
+                }));
+            }
+            return response.send(users);
+        });
+});
+
+usersRoute.post('/createInvestigator', (request: Request, response: Response) => {
```

Comment:
why do we neet this route?

Reply:
It was part of Lev's API so I created it too. If we dont need it then i will delete it